src/gui/properties/contentwidget.h

```diff
@@ -0,0 +1,145 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <memory>
+#include <QWidget>
+#include <QVector>
+#include <QString>
+#include <QMenu>
+
+#include "base/bittorrent/abstractfilestorage.h"
+#include "base/bittorrent/torrent.h"
+#include "base/bittorrent/torrentcontentlayout.h"
+#include "gui/torrentcontentfiltermodel.h"
+#include "gui/properties/proplistdelegate.h"
+#include "ui_contentwidget.h"
+
+class ContentWidget : public QWidget
+{
+    Q_OBJECT
+    Q_DISABLE_COPY_MOVE(ContentWidget)
+
+public:
+    explicit ContentWidget(QWidget *parent);
+    ~ContentWidget() override {};
+
+    // content widget can be backed either by an abstract file storage, or by a true torrent, and in
+    // the latter case extra features are enabled. In practice, AFS used in addnewtorrentdialog,
+    // torrent used in the properties panel.
+    void setContentAbstractFileStorage(BitTorrent::AbstractFileStorage *info, QStringList originalFilePaths);
+    void setContentTorrent(BitTorrent::Torrent *torrent);
+
+    QByteArray saveState() const;
+    void loadState(const QByteArray &state);
+    // number of bytes among the files selected to be downloaded.
+    qlonglong selectedSize() const;
+    // if the content widget is backed by a Torrent, update availability and progress
+    void loadDynamicData();
+    void contentLayoutChanged(const BitTorrent::TorrentContentLayout &);
+    QVector<BitTorrent::DownloadPriority> getFilePriorities() const;
+    void clear();
+
+private:
+    std::unique_ptr<QMenu> m_actionsMenu;
+    std::unique_ptr<PropListDelegate> m_treeViewDelegate;
+    std::unique_ptr<Ui::ContentWidget> m_ui;
+    std::unique_ptr<TorrentContentFilterModel> m_filterModel;
+    BitTorrent::AbstractFileStorage *m_fileStorage = nullptr;
+    BitTorrent::Torrent *m_torrent = nullptr;
+    QStringList m_originalFilePaths;
+    Utils::Fs::RenameList m_undoState;
+    bool m_hasUndo;
```

Comment:
Private member variables are usually declared at the very end of the private section which in turn goes in the end of the class.
Take a look at existing files for the order of the other sections: private slots and signals (not sure if there is one).